src/consensus/aft/raft.h

```diff
@@ -1295,6 +1341,12 @@ namespace aft
         return;
       }
 
+      if (retirement_committable_idx.has_value())
+      {
+        send_append_entries_response(from, AppendEntriesResponseType::FAIL);
+        return;
+      }
```

Comment:
This looks like a universal failure after we've received this node's RCI - is this correct? If the primary didn't receive our ACK and retransmits the uncommitted suffix (we have RCI but they send us RCI again), we shouldn't respond with a `FAIL`?

Reply:
You're right, the condition is incorrect should be `if (retirement_committable_idx.has_value() && r.idx > retirement_committable_idx)` I believe.